### ISSUE TITLE
reveal_type not run in "unreachable" branches

### ISSUE THREAD

**Issue by hjwp** [python/mypy#18423]
**Bug Report**

It looks like `reveal_type()` doesn't work in "unreachable" branches, ie places where `assert_never()` passes.

**To Reproduce**

```python
from typing import assert_never, reveal_type


def foo(b: bool):
    reveal_type(b)
    match b:
        case True:
            reveal_type(b)
            print("True")
        case False:
            reveal_type(b)
            print("False")
        case _:
            assert_never(b)
            reveal_type(b)  # this one does not print
```

(also at https://mypy-play.net/?mypy=latest&python=3.12&gist=d2646350c3ec4aa892763cdda7d32201)

**Actual Behavior**

```
❯ mypy thing.py
thing.py:5: note: Revealed type is "builtins.bool"
thing.py:8: note: Revealed type is "Literal[True]"
thing.py:11: note: Revealed type is "Literal[False]"
Success: no issues found in 1 source file
```

**Expected behaviour**

I would expect the 4th `reveal_type()` to print something like `Revealed type is Never` ?

(note that the `assert_never()` does "pass").

**Your Environment**

mypy 1.14.1 (compiled: yes)
Python 3.13.0 (also seen in 3.12)

**Comment by A5rocks** [comment]
This is intentional because mypy doesn't type check unreachable code. For instance, see also https://github.com/python/mypy/issues/18329.

**Comment by hjwp** [comment]
HI @A5rocks thanks for answering!  I was confused because I thought `assert_never()` used the type checker, so implicitly the unreachable branch must be being type checked, and now i understand, i think -- `assert_never()` only causes a typing failure when the branch is _not_ unreachable.

Shall we close this issue?

**Comment by A5rocks** [comment]
Tbh yeah this is tracked by that other issue.
